src/ssh_audit/ssh_audit.py

```diff
@@ -836,82 +839,129 @@ def process_commandline(out: OutputBuffer, args: List[str], usage_cb: Callable[.
+    parser.add_argument('-t', '--timeout', action="store", dest='timeout', default='5', type=int)
+    parser.add_argument('-v', '--verbose', action="store_true", dest='verbose', default=None)
+
+
+    # Add long options to the parser
+    parser.add_argument('--conn-rate-test', action="store", dest='conn_rate_test', default='0', type=int)
+    parser.add_argument('--dheat', action="store", dest='dheat', default='0', type=int)
+    parser.add_argument('--lookup', action="store", dest='lookup', default=None)
+    parser.add_argument('--no-colors', action="store_true", dest='no_colors', default=None)
+    parser.add_argument('--skip-rate-test', action="store_true", dest='skip_rate_test', default=None)
+    parser.add_argument('--threads', action="store", dest='threads', default='1', type=int)
+
+    try:
+        namespace = parser.parse_args()
+
+        if namespace.help is True:
             usage_cb(out)
-        elif o in ('-1', '--ssh1'):
-            aconf.ssh1 = True
-        elif o in ('-2', '--ssh2'):
-            aconf.ssh2 = True
-        elif o in ('-4', '--ipv4'):
-            aconf.ipv4 = True
-        elif o in ('-6', '--ipv6'):
-            aconf.ipv6 = True
-        elif o in ('-p', '--port'):
-            oport = a
-        elif o in ('-b', '--batch'):
+
+        aconf.host = namespace.host
+        host = namespace.host
+        port = namespace.port
+        aconf.ssh1 = namespace.ssh1
+        aconf.ssh2 = namespace.ssh2
+        aconf.ipv4 = namespace.ipv4
+        aconf.ipv6 = namespace.ipv6
+
+        aconf.json = namespace.json
+        if namespace.json_indent is True:
+            setattr(namespace, 'json', True)
+            aconf.json = namespace.json
+            aconf.json_print_indent = namespace.json_indent
+
+        if namespace.batch is True:
             aconf.batch = True
             aconf.verbose = True
-        elif o in ('-c', '--client-audit'):
-            aconf.client_audit = True
-        elif o in ('-j', '--json'):
-            if aconf.json:  # If specified twice, enable indent printing.
-                aconf.json_print_indent = True
-            else:
-                aconf.json = True
-        elif o in ('-v', '--verbose'):
+
+        aconf.client_audit = namespace.client_audit
+
+        ttime = namespace.timeout
+        if ttime != 5:
+            aconf.timeout = float(namespace.timeout)
+            aconf.timeout_set = True
+
+        if namespace.verbose is True:
             aconf.verbose = True
             out.verbose = True
-        elif o in ('-l', '--level'):
-            if a not in ('info', 'warn', 'fail'):
-                usage_cb(out, 'level {} is not valid'.format(a))
-            aconf.level = a
-        elif o in ('-t', '--timeout'):
-            aconf.timeout = float(a)
-            aconf.timeout_set = True
-        elif o in ('-M', '--make-policy'):
+
+        # Get error level regex
+        err_level = (namespace.level)
+        pattern = re.compile(r'info|warn|fail')
```

Comment:
An easier method for checking this would be: `if err_level in ["info", "warn", "fail"]:`

Reply:
Agree - and fairly sure I did attempt that; but was having difficulty with the string matching not working correctly for unknown reasons.

Reasons being either me ; or argparse weirdness that I don't understand.

I'll try again and see if I can make this cleaner